apps/hs/qua-server/src/Handler/Mooc/Admin/ReviewRequest.hs

```diff
@@ -0,0 +1,148 @@
+                                <*> taskRes
+                                <*> fmap toTime onlyBeforeRes
+  let widget = do
+        [whamlet|
+          #{extra}
+          ^{fvInput expertView}
+          ^{fvInput taskView}
+          ^{fvInput onlyBeforeView}
+          <input type=submit value="Send Mail" class="btn btn-default">
+        |]
+  return (params, widget)
+
+reviewRequest :: SendReviewParams -> Handler Html
+reviewRequest params = do
+  let whereTask = maybe [] (\tId -> [CurrentScenarioTaskId ==. tId]) (mtaskId params)
+  scenarios <- runDB $ selectList (whereTask ++ [
+                   CurrentScenarioGrade ==. Nothing
+                 , CurrentScenarioLastUpdate <. onlyBefore params
+               ]) []
+  statusTxt <-
+    if length scenarios > 0 then do
+      render <- getUrlRender
+      let toLink (Entity _ sc) = render $ SubmissionViewerR
+                                            (currentScenarioTaskId sc)
+                                            (currentScenarioAuthorId sc)
+      let scLinks = fmap toLink scenarios
+      case mexpertId params of
+        Just expertId -> do
+          mexpert <- runDB $ get expertId
+          case mexpert of
+            Just expert -> do
+              sendReviewRequestMail scLinks expert
+              return "Email sent..."
+            Nothing -> return "Expert not found"
+        Nothing -> do
+          experts <- selectExperts
+          _ <- forM experts $ \(Entity _ ex) -> sendReviewRequestMail scLinks ex
+          return "Emails sent..."
+    else
+      return "No scenarios to review. Email not sent."
+  let statusTxt' = statusTxt::Text
+  defaultLayout [whamlet|
+                   <p>#{ statusTxt' }
+                   <p><a onclick="window.history.back()" href="#">Back
+                |]
+
+sendReviewRequestMail :: [Text] -> User -> Handler ()
+sendReviewRequestMail scLinks expert =
+  case userEmail expert of
+    Just email -> do
+      let mailText = intercalate "\n\n" [
+                         "Dear " <> userName expert
+                       , "The following submissions are in need of reviewing:"
+                       , intercalate "\n" scLinks
+                       , "Thank you for your help!"
+                       ]
+      $(logDebug) mailText
+      liftIO $ Mail.renderSendMail $ Mail.simpleMail'
+          (Mail.Address Nothing email) --to address
+          (Mail.Address (Just "ETH qua-kit") "noreply@qua-kit.ethz.ch") --from
+          "Please help review the following submissions" --subject
+            $ fromStrict mailText
+    Nothing -> return ()
+
+selectExperts :: Handler [Entity User]
+selectExperts = runDB $
+  selectList [UserRole ==. UR_EXPERT, UserEmail /<-. [Nothing]] []
+
+selectTasks :: Handler [TaskData]
+selectTasks = runDB $ (map getVal) <$> rawSql query []
+  where
+    getVal (scp, Single ungradedCount) = (scp, ungradedCount)
+    query = unlines [
+            "SELECT ??, ungraded_count"
```

Comment:
This can be easily done using one sql query, (should be portable to esqueleto):
```
select scenario_problem.description, count(*) from scenario_problem, current_scenario
where current_scenario.task_id = scenario_problem.id and current_scenario.grade is null
group by current_scenario.task_id, scenario_problem.description;
```